app/Icepeak.hs

```diff
@@ -33,16 +40,28 @@ installHandlers core serverThread =
 
 main :: IO ()
 main = do
-  core <- Core.newCore
-  httpServer <- HttpServer.new core
-  let wsServer = WebsocketServer.acceptConnection core
-  pops <- Async.async $ Core.processOps core
-  upds <- Async.async $ WebsocketServer.processUpdates core
-  serv <- Async.async $ Server.runServer wsServer httpServer
-  logger <- Async.async $ processLogRecords (coreLogRecords core)
-  installHandlers core serv
-  log "System online. ** robot sounds **" (coreLogRecords core)
-  void $ Async.wait pops
-  void $ Async.wait upds
-  void $ Async.wait serv
-  void $ Async.wait logger
+  config <- execParser configInfo
+  -- load the persistent data from disk
+  let filePath = configDataFile config
+  eitherEncodedValue <- try $ withFile filePath ReadMode hGetContents
+
+  case (eitherEncodedValue :: Either SomeException ByteString) of
+      Left exc -> putStrLn $ "Failed to read the data from disk: " ++ show exc
+      Right encodedValue -> case eitherDecodeStrict encodedValue of
+          Left msg  -> error $ "Failed to decode the initial data: " ++ show msg
+          Right value -> do
```

Comment:
I think it would be slightly cleaner to extract this into a function `readValue :: FilePath -> IO (Either String Value)`, also to avoid rightward drift in `main`.

Reply:
Good point, I extracted it.